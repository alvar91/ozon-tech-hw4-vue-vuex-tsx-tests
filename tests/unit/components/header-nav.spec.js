import Vuex from "vuex";
import { mount, createLocalVue } from "@vue/test-utils";

import router from "@/router";

import HeaderNav from "@/components/header-nav.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("HeaderNav.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(HeaderNav, {
      localVue,
      router,
    });
  });

  it("should correctly renders markup", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  afterEach(() => {
    wrapper.destroy();
  });
});
