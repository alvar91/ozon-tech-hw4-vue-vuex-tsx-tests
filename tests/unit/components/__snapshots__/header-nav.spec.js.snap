// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNav.vue should correctly renders markup 1`] = `
<nav>
  <ul class="nav__list">
    <li class="nav__item"><a href="/" aria-current="page" class="router-link-exact-active nav__link--active nav__link">Все</a></li>
    <li class="nav__item"><a href="/waiting" class="nav__link">Ожидают загрузки</a></li>
    <li class="nav__item"><a href="/loaded" class="nav__link">Загружены кандидатом</a></li>
  </ul>
</nav>
`;
